src/garage/_dtypes.py

```diff
@@ -559,3 +559,214 @@ def output_space(self):
 
         """
         return self._output_space
+
+
+class SampleBatch(
+        collections.namedtuple('SampleBatch', [
+            'observations', 'actions', 'rewards', 'next_observations',
+            'terminals', 'batch_size'
+        ])):
+    # pylint: disable=missing-param-doc, missing-type-doc
+    """A tuple representing a batch of samples.
+
+    Attributes:
+        observations (numpy.ndarray): Non-flattened array of observations.
+            Typically has shape (batch_size, S^*) (the unflattened state space
+            of the current environment).
+        actions (numpy.ndarray): Non-flattened array of actions. Should
+            have shape (batch_size, S^*) (the unflattened action space of the
+            current environment).
+        rewards (numpy.ndarray): Array of rewards of shape (batch_size,) (1D
+            array of length batch_size).
+        next_observation (numpy.ndarray): Non-flattened array of next
+            observations. Has shape (batch_size, S^*). next_observations[i] was
+            observed by the agent after taking actions[i].
+        terminals (numpy.ndarray): A boolean numpy array of shape
+            shape (batch_size,) containing the termination signals for all
+            transitions in this batch.
+        batch_size (int): The size of the batch
```

Comment:
this is inferrable from len(terminals) (or anything else) so it doesn't need to be included.

Reply:
If we remove batch_size, we would assume that all param comes with the same length. In other words we can't verify the length of inputs, and `TimeStepBatch` will rely on its producer to pass in arrs with equal length. Is this acceptable?

Reply:
take a look at what TrajectoryBatch does.

basically, it verifies all inputs are the same length